test/parallel/test-repl-save-load.js

```diff
@@ -20,162 +20,189 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
+
 const common = require('../common');
 const ArrayStream = require('../common/arraystream');
-const assert = require('assert');
-const fs = require('fs');
+
+const assert = require('node:assert');
+const fs = require('node:fs');
+const repl = require('node:repl');
+const path = require('node:path');
+
 const tmpdir = require('../common/tmpdir');
 tmpdir.refresh();
 
-// TODO: the following async IIFE and the completePromise function are necessary because
-//       the reply tests are all run against the same repl instance (testMe) and thus coordination
-//       needs to be in place for the tests not to interfere with each other, this is really
-//       not ideal, the tests in this file should be refactored so that each use its own isolated
-//       repl instance, making sure that no special coordination needs to be in place for them
-//       and also allowing the tests to all be run in parallel
-(async () => {
-  const repl = require('repl');
-
-  const works = [['inner.one'], 'inner.o'];
-
-  const putIn = new ArrayStream();
-  const testMe = repl.start('', putIn);
-
-  // Some errors might be passed to the domain.
-  testMe._domain.on('error', function(reason) {
-    const err = new Error('Test failed');
-    err.reason = reason;
-    throw err;
+function prepareREPL(replOpts = {}) {
+  const input = new ArrayStream();
+  const output = new ArrayStream();
+
+  const replServer = repl.start({
+    prompt: '',
+    input,
+    output,
+    allowBlockingCompletions: true,
+    ...replOpts,
   });
 
-  async function completePromise(query, callback) {
-    return new Promise((resolve) => {
-      testMe.complete(query, (...args) => {
-        callback(...args);
-        resolve();
-      });
-    });
-  }
+  // Some errors are passed to the domain, but do not callback
+  replServer._domain.on('error', assert.ifError);
+
+  return { replServer, input, output };
+}
 
-  const testFile = [
+// The tests in this file test the REPL saving and loading session data to/from a file
+
+
+// The REPL can save a session's data to a file and load it back
+{
+  const { replServer, input } = prepareREPL();
+  const tmpDirPath = tmpdir.random();
```

Comment:
Can't we refresh again instead of adding `tmpdir.random()`?

Reply:
Yes this is my bad, thanks for pointing it out :pray:

I was planning to have the tests run in parallel and wanted to make sure that they would not all use the same directory, let me revert the `random` change :pray: